deposit/resource.ts

```diff
@@ -152,3 +177,86 @@ export const getIdentifier = (resource: AnyResource, identifierKind: ResourceIde
 export const getIdentifierValue = (resource: AnyResource, identifierKind: ResourceIdentifierKind) =>
 	resource.identifiers.find((identifier) => identifier.identifierKind === identifierKind)
 		?.identifierValue;
+
+export const partialResourceSchema = z.object({
```

Comment:
i made this before i had the `/utils/api/schemas/` convention, maybe should be moved there.

Also, might make more sense to define the types based on the schema now, but its not that important